build/common/installer/scripts/fluent-bit-conf-customizer.rb

```diff
@@ -35,6 +43,20 @@ def substituteMultiline(multilineLogging, stacktraceLanguages, new_contents)
     return new_contents
 end
 
+def substituteResourceOptimization(resoureceOptimizationEnabled, new_contents)
+  if (!resoureceOptimizationEnabled.nil? && resoureceOptimizationEnabled.to_s.downcase == "true") || (@isWindows && @using_aad_msi_auth)
+    puts "config::Starting to substitute the placeholders in fluent-bit.conf file for resource optimization"
+    if (@isWindows && @using_aad_msi_auth)
+      new_contents = new_contents.gsub("#${ResourceOptimizationPluginFile}", "plugins_file  /etc/fluent-bit/azm-containers-input-plugins.conf")
+    else
+      new_contents = new_contents.gsub("#${ResourceOptimizationPluginFile}", "plugins_file  /etc/opt/microsoft/docker-cimprov/azm-containers-input-plugins.conf")
```

Comment:
as discussed, if the resoureceOptimizationEnabled = true and legacy, then we are updating linux conf file path - /etc/opt/microsoft/docker-cimprov/azm-containers-input-plugins.conf for windows. This is bug.